File: backend/src/main/java/com/bookstacker/service/BookService.java
package com.bookstacker.service;

import com.bookstacker.model.Book;
import com.bookstacker.repository.BookRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class BookService {
    private final BookRepository bookRepository;

    public BookService(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    public List<Book> getAllBooks() {
        return bookRepository.findAll();
    }

    public Book getBookById(Long id) {
        return bookRepository.findById(id).orElseThrow();
    }

    public Book createBook(Book book) {
        return bookRepository.save(book);
    }

    public Book updateBook(Long id, Book updatedBook) {
        Book book = bookRepository.findById(id).orElseThrow();
        book.setTitle(updatedBook.getTitle());
        book.setAuthor(updatedBook.getAuthor());
        book.setGenre(updatedBook.getGenre());
        book.setDescription(updatedBook.getDescription());
        book.setPublishedDate(updatedBook.getPublishedDate());
        return bookRepository.save(book);
    }

    public void deleteBook(Long id) {
        bookRepository.deleteById(id);
    }
}

File: frontend/src/services/BookService.js
import axios from 'axios';

const BASE_URL = 'http://localhost:8080/api/books';

class BookService {
  getBooks() {
    return axios.get(BASE_URL);
  }

  getBookById(id) {
    return axios.get(`${BASE_URL}/${id}`);
  }

  createBook(book) {
    return axios.post(BASE_URL, book);
  }

  updateBook(id, book) {
    return axios.put(`${BASE_URL}/${id}`, book);
  }

  deleteBook(id) {
    return axios.delete(`${BASE_URL}/${id}`);
  }
}

export default new BookService();